### PR TITLE
Sending category, description and session id to splunk and generating…

### PR DESCRIPTION
**Pull request by HEskandari**
Connects to https://github.com/Particular/EndToEnd/issues/3

Sending category, description and session id to splunk and generating the start batch file

### CODE REVIEW

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 1]
```diff
@@ -95,11 +101,15 @@ static void DeleteOtherPlatformHost(Permutation permutation)
...
+    }
+    
+    [SetUpFixture]
+    public class TestSessionInitializer
+    {
+        [SetUp]
+        public void Setup()
         {
-            SessionId = Guid.NewGuid().ToString();
+            Base.SessionId = DateTime.UtcNow.Ticks.ToString();
```
Comment (ramonsmits) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Why share the same session ID across *all* test? How can I filter on a single run?

Comment (ramonsmits) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Also, trying to understand this:

>Attribute used to mark a class that contains one-time SetUp and/or TearDown methods that apply to all the tests in a namespace or an assembly.

Does this mean a new session ID per fixture, or for *all* fixtures?

File: src/PerformanceTests/Tests/Statistics.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
+﻿using System;
```
Comment (ramonsmits) [inline src/PerformanceTests/Tests/Statistics.cs]: Why is this moved to the test project? It already is becoming the root of all evil. Stuff should be moved out of it.

Comment (ramonsmits) [inline src/PerformanceTests/Tests/Statistics.cs]: Also, you didn't move / rename the file as it is mared as "Added" in git, loosing history on its change

File: src/PerformanceTests/NServiceBus6/NServiceBus6.csproj [hunk 3]
```diff
@@ -80,10 +80,6 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\Common\Utils.csproj">
```
Comment (ramonsmits) [inline src/PerformanceTests/NServiceBus6/NServiceBus6.csproj]: You Removed the Utils project from the solution, but kept the folder.

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 4]
```diff
@@ -39,10 +39,14 @@ public virtual void GatedPublishRunner(Permutation permutation)
 
         void Tasks(Permutation permutation, [CallerMemberName] string memberName = "")
         {
-            permutation.Category = GetType().Name;
+            var fixture = GetType().GetCustomAttribute<TestFixtureAttribute>();
+            permutation.Category = fixture.Category;
+            permutation.Description = fixture.Description;
```
Comment (ramonsmits) [inline src/PerformanceTests/Tests/Categories/Base.cs]: What is the benefit of passing a description all the way to the NLog global context and add it to each and every line?

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 5]
```diff
@@ -61,17 +65,19 @@ static void LaunchAndWait(Permutation permutation)
             var sessionIdArgument = string.Format(" --sessionId={0}", SessionId);
 
             var exe = new FileInfo(permutation.Exe);
+            var arguments = PermutationParser.ToArgs(permutation) + processIdArgument + sessionIdArgument;
```
Comment (dvdstelt) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Can't you do something like
```
var permutationArguments = PermutationParser.ToArgs(permutation);
var arguments = string.Format("{0} {1} {2}", permutationArguments, processIdArgument, sessionIdArgument);
```
I'd had to look at it carefully to see what the intention was there. Because PermutationParser.ToArgs() feels like a list is being returned, not a single string. So that concatenation wasn't clear to me.

File: src/PerformanceTests/Tests/Permutations/PermutationParser.cs [hunk 6]
```diff
@@ -10,17 +10,24 @@ public static class PermutationParser
 
         public static string ToArgs(Permutation instance)
         {
-            return "--tests:" + string.Join(Seperator, instance.Tests) + " --variables:" + ToString(instance);
+            return "--tests:" + string.Join(Seperator, instance.Tests) + 
+                   " --category:" + instance.Category +
+                   " --description:\"" + instance.Description + "\"" +
+                   " --variables:" + ToString(instance);
```
Comment (dvdstelt) [inline src/PerformanceTests/Tests/Permutations/PermutationParser.cs]: Why a : here, but a = in the other arguments? The parser will handle both, but = seems more consistent?

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 7]
```diff
@@ -61,17 +65,19 @@ static void LaunchAndWait(Permutation permutation)
...
 
-            var pi = new ProcessStartInfo(exe.FullName, PermutationParser.ToArgs(permutation) + processIdArgument + sessionIdArgument)
+            var pi = new ProcessStartInfo(exe.FullName)
             {
+                Arguments = arguments,
```
Comment (ramonsmits) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Don't use the arguments property while there is constructor overload that accepts arguments.

File: src/PerformanceTests/Tests/Permutations/PermutationParser.cs (hunk 6 above)
Comment (ramonsmits) [inline src/PerformanceTests/Tests/Permutations/PermutationParser.cs]: @dvdstelt The = is used for the variable value, not for the values of the arguments.

```
/tests:MySuperTest /variables:Audit=Off;DTC=On
```

**Comment by ramonsmits** [comment]
So instead of moving stuff to Tests I would be in favor of moving stuff to Utils. The Utils should probably be renamed to something like Core. The current dependency chain where Tests is the root was a design flaw.

File: src/PerformanceTests/Tests/Categories/Base.cs (hunk 1 above)
Comment (HEskandari) [inline src/PerformanceTests/Tests/Categories/Base.cs]: This is to generate a build ID if you will, where we run all the tests. To have a Test or Fixture ID we can have the same approach but generate them per test. What I wanted to do here was to separate latest 'batch' of tests from all the others builds/test run on the same day or previous days.

Comment (HEskandari) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Also you can filter them by category and description + latest run. e.g. show me the test result of the latest build where test category is performance or test description is 'Audit forwarding On / Off'.

File: src/PerformanceTests/Tests/Statistics.cs (hunk 2 above)
Comment (HEskandari) [inline src/PerformanceTests/Tests/Statistics.cs]: Need to read stuff off Permutation object. So this can go where that is. There was hardly any benefit of having a project with just Statistics in it.

File: src/PerformanceTests/Tests/Categories/Base.cs (hunk 4 above)
Comment (HEskandari) [inline src/PerformanceTests/Tests/Categories/Base.cs]: We could use the test name instead, but the point was to use it as a filter if we need to filter out the result.
